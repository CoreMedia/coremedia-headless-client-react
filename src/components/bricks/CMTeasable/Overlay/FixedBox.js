// @flow
import styled from 'styled-components';
import { media, getColors } from '../../../../styles/themes/utils';

const FixedBox = styled.div`
  color: ${props => getColors(props.theme.colors, props.color).fg};
  background-color: ${props => getColors(props.theme.colors, props.color).bgtrans};
  overflow: hidden;
  padding: 1em;
  max-width: 100%;
  max-height: 100%;
  position: absolute;
  width: ${props => props.width}%;
  left: ${props => (100 - props.width) / 2}%;
  bottom: ${props => props.bottom};
  ${props => media(props.theme.breakpoints.tablet)`height: ${props => props.height};`};
`;

FixedBox.displayName = 'FixedBox';

FixedBox.defaultProps = {
  theme: {
    colors: {
      red: {
        fg: '#fafafa',
        bgsolid: 'rgba(221, 52, 40, 1)',
        bgtrans: 'rgba(221, 52, 40, 0.7)',
        huerotation: '320deg',
      },
      default: {
        fg: '#fafafa',
        bgsolid: 'rgba(239, 223, 15, 1)',
        bgtrans: 'rgba(239, 223, 15, 0.7)',
        huerotation: '5deg',
      },
    },
    breakpoints: {
      tablet: 768,
    },
  },
};

export default FixedBox;
