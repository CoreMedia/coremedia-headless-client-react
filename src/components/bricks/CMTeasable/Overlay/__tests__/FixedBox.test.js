// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import FixedBox from '../FixedBox';

describe('FixedBox Component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <FixedBox width={70} height="100%" bottom="10px">
          Content
        </FixedBox>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
