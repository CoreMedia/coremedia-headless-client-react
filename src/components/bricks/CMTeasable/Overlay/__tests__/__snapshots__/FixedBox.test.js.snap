// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedBox Component should render correctly 1`] = `
.c0 {
  color: #fafafa;
  background-color: rgba(239,223,15,0.7);
  overflow: hidden;
  padding: 1em;
  max-width: 100%;
  max-height: 100%;
  position: absolute;
  width: 70%;
  left: 15%;
  bottom: 10px;
}

@media (min-width:48em) {
  .c0 {
    height: 100%;
  }
}

<div
  className="c0"
  height="100%"
  width={70}
>
  Content
</div>
`;
